FHEM/14_SD_WS_Maverick.pm

```diff
@@ -80,112 +92,202 @@ SD_WS_Maverick_Parse($$)
- 
-    readingsEndUpdate($hash, 1); # Notify is done by Dispatch
+  }
+
+  $hash->{lastReceive} = time();
+  $hash->{lastMSG} = $rawData;
+  #$hash->{bitMSG} = $bitData2; 
+  
+  # Die Temperaturen bei Inaktivität zurücksetzen lassen durch Timer ...
+  my $inaktivityInterval=int(AttrVal($name,"inaktivityInterval",6));
+  if ($temp_bbq ne "") {
+      RemoveInternalTimer($hash, 'SD_WS_Maverick_ClearTempBBQ');
+      InternalTimer(time()+($inaktivityInterval*60), 'SD_WS_Maverick_ClearTempBBQ', $hash, 0);
+  }
+  if ($temp_food ne "") {
+      RemoveInternalTimer($hash, 'SD_WS_Maverick_ClearTempFood');
+      InternalTimer(time()+($inaktivityInterval*60), 'SD_WS_Maverick_ClearTempFood', $hash, 0);
+  }
+  
+  # Checksum auswerten
+  $checksum_str =~ tr/569A/0123/;
+  my $checksum="";
+  $checksum=$checksum_str; 
+  # TODO: Die eigentliche Checksum errechnen. Diese ändert sich bei jedem Temperaturwechsel
+  # TODO: Evtl. ist in den checksum-bits auch noch eine Info zur Batterie enthalten
+  #       ggf. ist es möglich die checksum als ID zu verwenden und so mehrere Mavericks in fhem einbinden zu können.
+  $hash->{checksum}=$checksum;
+  
+  # TODO: Logging kann entfernt werden, wenn checksum entschlüsselt ist. Wird zur Analyse verwendet.
+  Log3 $hash, 4, "$name statistic: checksum=$checksum, t1=$temp_str1, temp-food=$temp_food, t2_$temp_str2, temp-bbq=$temp_bbq;";
+  
+  SD_WS_Maverick_updateReadings($hash, $temp_food, $temp_bbq, $startup);
 
-	return $name;
+  return $name;
 
 }
 
 sub SD_WS_Maverick_Attr(@)
 {
-  my @a = @_;
-
-  # Make possible to use the same code for different logical devices when they
-  # are received through different physical devices.
-  return  if($a[0] ne "set" || $a[2] ne "IODev");
-  my $hash = $defs{$a[1]};
-  my $iohash = $defs{$a[3]};
-  my $cde = $hash->{CODE};
-  delete($modules{SD_WS_Maverick}{defptr}{$cde});
-  $modules{SD_WS_Maverick}{defptr}{$iohash->{NAME} . "." . $cde} = $hash;
+  my ($cmd,$name,$attr_name,$attr_value) = @_;
+  my $hash = $defs{$name};
+  if($cmd eq "set") {
+    if($attr_name eq "IODev") {
+      # Make possible to use the same code for different logical devices when they
+      # are received through different physical devices.
+      my $iohash = $defs{$attr_value};
+      my $cde = $hash->{CODE};
+      delete($modules{SD_WS_Maverick}{defptr}{$cde});
+      $modules{SD_WS_Maverick}{defptr}{$iohash->{NAME} . "." . $cde} = $hash;
+    }
+    elsif($attr_name eq "inaktivityInterval") {
+      if (!looks_like_number($attr_value) || int($attr_value) < 1 || int($attr_value) > 60) {
+          return "$name: Value \"$attr_value\" is not allowed.\n"
+                 ."inaktivityInterval must be a number between 1 and 60."
+      }
+    }
+  }
   return undef;
 }
 
+sub SD_WS_Maverick_ClearTempBBQ($){
+  my ($hash) = @_;
+  my $name = $hash->{NAME};
+  Log3 $hash, 5, "$name ClearTempBBQ";
+  
+  my $temp_bbq=-1;
+  my $temp_food=ReadingsVal($name,"temp-food",-1);
```

Comment:
Wieso spendierst Du nicht ein eigenes Reading für den Lebens Zustand?

Wenn ich mir jetzt die Werte der Sensoren loggen würde und diese dann später in einem Plot darstelle, könnte es sein, dass mein Plot bescheiden aussieht, wenn ich Empfangsprobleme habe.

Reply:
Hierüber muss ich etwas länger nachgrübeln.

Reply:
Baue dann doch je Sensor einen state ein mit den Werten connected, disconnected oder inactiv. Dadurch werden notifies evtl. Etwas komplizierter, aber auch verständlicher. Und wenn man die einmal fertig hat kann man in den plots stolz störungsfreie Plateauphasen des PulledPork zeigen ;) Das ist denke ich wichtiger als initial einmal etwas weniger Aufwand zu haben. 

Komme hoffentlich heute Abend dazu. Ansonsten erst übernächste Woche.

Reply:
neues Reading für beide Sensoren (Sensor-1-food_state; Sensor-2-bbq_state) mit den möglichen Werten connected, disconnectet oder inactiv.